### DISCUSSION TITLE
How to put the running number by exclude the invisible columns

### DISCUSSION THREAD

**Discussion by ShermaineMY** [eclipse-birt/birt#1017] [Q&A]
I am using BIRT Designer. I have create a table. Each row has different cell with different secondary data set. If that row with empty value, that row will auto hidden. 

I wish to add to number at first column. How to put the correct running number? The correct running number by without count the hidden row.

Thanks

**Comment by hvbtup** [comment]
It may be possible, but it's certainly tricky.

The technical issue is probably that filtering and counting must happen in the outer table.
But to see if filtering actually should happen, you need the data from the inner table, which is "invisible" for the outer table.

How are the data sets related logically and how many datasets are we talking about?
If there are only 2 or 3 data sets, consider using "Joint Data Sets" (you'll need n-1 Joint data sets for n data sets).
Then your table can bind to the "outer-most" of these joint data sets.
Inside this joint data set, you can access the data from all data sets, which makes filtering easier.

I wonder what the use case for such a data model structure is...

You mention: "Each row has different cell....".
Do you mean: one cell (e.g. a table with only one column?) Then you should consider using a List Item instead of a Table Item.

You can put all the different detail data into the detail area of that List Item.
Or, if you still want to use a table: put all the different detail data into the same cell of the outer Table Item.
That way there is no need to hide the row - instead you just hide the inner layout object if it contains no data.
